homeassistant/components/iometer/sensor.py

```diff
@@ -0,0 +1,160 @@
+"""IOmeter sensors."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    PERCENTAGE,
+    SIGNAL_STRENGTH_DECIBELS,
+    STATE_UNKNOWN,
+    EntityCategory,
+    UnitOfEnergy,
+    UnitOfPower,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .coordinator import IOMeterCoordinator, IOmeterData
+from .entity import IOmeterEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class IOmeterEntityDescription(SensorEntityDescription):
+    """Describes IOmeter sensor entity."""
+
+    value_fn: Callable[[IOmeterData], str | int | float]
+
+
+SENSOR_TYPES: list[IOmeterEntityDescription] = [
+    IOmeterEntityDescription(
+        key="meter_number",
+        translation_key="meter_number",
+        icon="mdi:meter-electric",
+        value_fn=lambda data: data.status.meter.number,
+    ),
+    IOmeterEntityDescription(
+        key="connection_status",
+        translation_key="connection_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=["connected", "disconnected"],
+        value_fn=lambda data: data.status.device.core.connection_status,
+    ),
+    IOmeterEntityDescription(
+        key="wifi_rssi",
+        translation_key="wifi_rssi",
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda data: data.status.device.bridge.rssi,
+    ),
+    IOmeterEntityDescription(
+        key="core_bridge_rssi",
```

Comment:
can we set `entity_registry_enabled_default=False` on this one?

Reply:
Same for this sensor, what would the benefit be?

Reply:
No the RSSI's are not used a lot. Therefore lets disable them by default.